docs/data/data-grid/overlays/LoadingOverlayVariants.js

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import FormControl from '@mui/material/FormControl';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import InputLabel from '@mui/material/InputLabel';
+import Switch, { SwitchProps } from '@mui/material/Switch';
+import Select, { SelectProps } from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import { DataGrid, GridLoadingOverlayVariant } from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VariantControl = (props) => {
+  const { value, onChange, label, id } = props;
+  const labelId = `${id}-label`;
+  return (
+    <FormControl>
+      <InputLabel htmlFor={id} id={labelId}>
+        {label}
+      </InputLabel>
+      <Select
+        label={label}
+        id={id}
+        labelId={labelId}
+        value={value}
+        onChange={onChange}
+        size="small"
+        sx={{ width: 180 }}
+      >
+        <MenuItem value="circular-progress">Circular Progress</MenuItem>
+        <MenuItem value="linear-progress">Linear Progress</MenuItem>
+        <MenuItem value="skeleton">Skeleton</MenuItem>
+      </Select>
+    </FormControl>
+  );
+};
+
+const RowsControl = (props) => {
+  const { checked, onChange } = props;
+  return (
+    <FormControlLabel
+      control={<Switch checked={checked} onChange={onChange} />}
+      label="Rows"
+    />
+  );
+};
+
+export default function LoadingOverlayVariants() {
+  const [withRows, setWithRows] = React.useState(true);
+  const [variant, setVariant] =
+    React.useState < GridLoadingOverlayVariant > 'linear-progress';
+  const [noRowsVariant, setNoRowsVariant] =
+    React.useState < GridLoadingOverlayVariant > 'skeleton';
```

Comment:
Something went wrong here 😅
There's a `docs:typescript:formatted` script that generates `.js` demo files from `.tsx` – this should help.

Reply:
Got it, thanks!